numpy/random/tests/test_regression.py

```diff
@@ -113,5 +113,20 @@ def test_choice_sum_of_probs_tolerance(self):
             assert_(c in a)
             assert_raises(ValueError, np.random.choice, a, p=probs*0.9)
 
+    def test_permutation_of_array_of_different_length_strings(self):
+        # Test that permuting an array of different length strings
+        # will not cause a segfault on garbage collection
+        # Tests gh-7710
+        np.random.seed(1234)
+
+        a = np.array(['a', 'a' * 1000])
+
+        for _ in range(100):
+            z = np.random.permutation(a)
+
+        # Force Garbage Collection - should not segfault.
```

Comment:
Is explicit garbage collection really necessary? Everything seems like it should be cleaned up on function exit anyway (no obvious reference loops or so?).